src/main/java/com/microsoft/sqlserver/jdbc/SQLServerColumnEncryptionCertificateStoreProvider.java

```diff
@@ -74,11 +101,123 @@ public String getName() {
+        }  else {
             throw new SQLServerException(SQLServerException.getErrString("R_notSupported"), null);
         }
         windowsCertificateStoreLogger.exiting(SQLServerColumnEncryptionCertificateStoreProvider.class.getName(),
                 "decryptColumnEncryptionKey", "Finished decrypting Column Encryption Key.");
         return plainCek;
     }
+
+    private byte[] decryptColumnEncryptionKeyNonWindows(String masterKeyPath,
+                                                        String encryptionAlgorithm,
+                                                        byte[] encryptedColumnEncryptionKey) throws SQLServerException {
+        KeyStoreProviderCommon.validateNonEmptyMasterKeyPath(masterKeyPath);
+        CertificateDetails certificateDetails = getCertificateDetailsNonWin(masterKeyPath);
+        return KeyStoreProviderCommon.decryptColumnEncryptionKey(masterKeyPath, encryptionAlgorithm,
+                encryptedColumnEncryptionKey,
+                certificateDetails);
+    }
+
+    private CertificateDetails getCertificateDetailsNonWin(String masterKeyPath) throws SQLServerException {
+        String storeLocation = null;
+
+        String[] certParts = masterKeyPath.split("/");
+
+        // Validate certificate path
+        // Certificate path should only contain 3 parts (Certificate Location, Certificate Store Name and Thumbprint)
+        if (certParts.length > 3) {
+            throw new SQLServerException(SQLServerException.getErrString("R_AECertpathBad"), null);
+        }
+
+        // Extract the store location where the cert is stored
+        if (certParts.length > 2) {
+            if (certParts[0].equalsIgnoreCase(localMachineDirectory)) {
+                storeLocation = localMachineDirectory;
+            }
+            else if (certParts[0].equalsIgnoreCase(currentUserDirectory)) {
+                storeLocation = currentUserDirectory;
+            }
+            else {
+                throw new SQLServerException(SQLServerException.getErrString("R_AECertLocBad"), null);
+            }
+        }
+
+        // Parse the certificate store name. Only store name "My" is supported.
+        if (certParts.length > 1) {
+            if (!certParts[certParts.length - 2].equalsIgnoreCase(myCertificateStore)) {
+                throw new SQLServerException(SQLServerException.getErrString("R_AECertLocBad"), null);
+            }
+        }
+
+        // Get thumpbrint
+        String thumbprint = certParts[certParts.length - 1];
+        if ((null == thumbprint) || (0 == thumbprint.length())) {
+            // An empty thumbprint specified
+            throw new SQLServerException(SQLServerException.getErrString("R_AECertHashEmpty"), null);
+        }
+
+        // Find the certificate and return
+        return getCertificateByThumbprintNonWin(thumbprint, masterKeyPath);
+    }
+
+    private String getThumbPrint(X509Certificate cert) throws NoSuchAlgorithmException, CertificateEncodingException {

```

Comment:
**Reporter**: CHECKSTYLE
**Rule**: com.puppycrawl.tools.checkstyle.checks.sizes.LineLengthCheck
**Severity**: ERROR
**File**: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerColumnEncryptionCertificateStoreProvider.java L166

Line is longer than 80 characters (found 118).

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<-1054340309>*